File: lib/model/keberangkatan.dart
import 'dart:convert';
import 'package:SempatiStar.app/model/lambung.dart';
import 'package:SempatiStar.app/model/trayek.dart';

class Keberangkatan {
  Lambung lambung;
  String jam;
  String waktu;
  Trayek trayek;
  int id;
  String tanggal;
  Null armada;

  Keberangkatan({
  this.lambung,
  this.jam = "",
  this.waktu = "",
  this.trayek,
  this.id = 0,
  this.tanggal = "",
  this.armada = null,
  });

  Keberangkatan.fromJson(Map<String, dynamic>  map) :
        lambung = map['lambung'] == null
            ? null
            : Lambung.fromJson(map['lambung']),
        jam = map['jam']  ?? "",
        waktu = map['waktu']  ?? "",
        trayek = map['trayek'] == null
            ? null
            : Trayek.fromJson(map['trayek']),
        id = map['id']  ?? 0,
        tanggal = map['tanggal']  ?? "",

  Map<String, dynamic> toJson() => {
        'lambung': lambung.toJson(),
        'jam': jam,
        'waktu': waktu,
        'trayek': trayek.toJson(),
        'id': id,
        'tanggal': tanggal,
        'armada': armada,
      };

  Keberangkatan copyWith({
    Lambung lambung,
    String jam,
    String waktu,
    Trayek trayek,
    int id,
    String tanggal,
    Null armada,
  }) {
    return Keberangkatan(
      lambung: lambung ?? this.lambung,
      jam: jam ?? this.jam,
      waktu: waktu ?? this.waktu,
      trayek: trayek ?? this.trayek,
      id: id ?? this.id,
      tanggal: tanggal ?? this.tanggal,
      armada: armada ?? this.armada,
    );
  }
}

File: app/src/main/java/sempati/star/app/models/Keberangkatan.java
package sempati.star.app.models;

public class Keberangkatan {
    private float id;
    private String tanggal;
    private String jam;
    private String waktu;
    Lambung LambungObject;
    TbTrayek TrayekObject;
    private String armada = null;


    // Getter Methods

    public float getId() {
        return id;
    }

    public String getTanggal() {
        return tanggal;
    }

    public String getJam() {
        return jam;
    }

    public String getWaktu() {
        return waktu;
    }

    public Lambung getLambung() {
        return LambungObject;
    }

    public TbTrayek getTrayek() {
        return TrayekObject;
    }

    public String getArmada() {
        return armada;
    }

    // Setter Methods

    public void setId(float id) {
        this.id = id;
    }

    public void setTanggal(String tanggal) {
        this.tanggal = tanggal;
    }

    public void setJam(String jam) {
        this.jam = jam;
    }

    public void setWaktu(String waktu) {
        this.waktu = waktu;
    }

    public void setLambung(Lambung lambungObject) {
        this.LambungObject = lambungObject;
    }

    public void setTrayek(TbTrayek trayekObject) {
        this.TrayekObject = trayekObject;
    }

    public void setArmada(String armada) {
        this.armada = armada;
    }
}
